src/lib/components/Map/Map.js

```diff
@@ -105,10 +105,23 @@ class Map extends Component {
   addMouseEvents(mapId) {
     const { handlers, APP } = this.props;
     if (handlers && Array.isArray(handlers)) {
-      // let handler;
-
+      let handler;
+      let sublayer;
       for (let c = 0; c < handlers.length; c += 1) {
-        this.map.on(handlers[c].type, handlers[c].method);
+        handler = handlers[c];
+        if (handler.layer) {
+          if (handler.layer.layers) {
```

Comment:
@KipSigei we should check first if `handler.layer` is a string

Reply:
@cKellyDesignI don't think we need that check because we will be passing the entire active layer object into the `on` instance so we will instead have `handler.layer.id` https://github.com/onaio/gisida-react/pull/102/commits/7c84a0f79e5365ed5b3ba1f0953cd078327a6046, for the active grouped layer object we will just check the sublayer ids array.

Reply:
Aaaaaah ok, pole i thought we were just passing the string, but that makes sense!